source/workers/LocalWorker.cpp

```diff
@@ -3867,37 +3908,206 @@ void LocalWorker::s3ModeIterateCustomObjects()
+
+#endif // S3_SUPPORT
+}
+
+/**
+ * Create given S3 bucket.
+ *
+ * @throw WorkerException on error.
+ */
+void LocalWorker::s3ModeHeadBucket(std::string bucketName)
+{
+#ifndef S3_SUPPORT
+	throw WorkerException(std::string(__func__) + " called, but this was built without S3 support");
+#else
+	S3::HeadBucketRequest request;
+	request.SetBucket(bucketName);
+
+	S3::HeadBucketOutcome outcome = s3Client->HeadBucket(request);
+
+	if(!outcome.IsSuccess() )
+	{
+		auto s3Error = outcome.GetError();
+
+        throw WorkerException(std::string("Bucket creation failed. ") +
 				"Endpoint: " + s3EndpointStr + "; "
 				"Bucket: " + bucketName + "; "
 				"Exception: " + s3Error.GetExceptionName() + "; " +
 				"Message: " + s3Error.GetMessage() );
-		}
 	}
 
 #endif // S3_SUPPORT
 }
 
+/**
+ * Create tags for given S3 bucket.
+ *
+ * @throw WorkerException on error.
+ */
+void LocalWorker::s3ModeCreateBucketTagging(std::string bucketName)
+{
+#ifndef S3_SUPPORT
+    throw WorkerException(std::string(__func__) + " called, but this was built without S3 support");
+#else
+    // tag buckets
+    {
+/*        TODO:
+          verify no tags exist after delete tags (how to return tagset from GetBucketTagging)
+*/
+        const bool doS3TagInlineVerify = progArgs->getDoS3BucketTagInlineVerify();
+
+        S3::PutBucketTaggingRequest request;
+        S3::Tagging tagging;
+        S3::Tag tag;
+
+        std::vector<std::pair<std::string, std::string>> keyValuePairs = {
+                {TAG_KEY_SHORT_NAME, TAG_VALUE_SHORT},
+                {TAG_KEY_MEDIUM_NAME, StringTk::generateRandomS3TagValue(TAG_KEY_MEDIUM_NAME,
+                                                                         TAG_VALUE_MEDIUM_LEN,
+                                                                         TAG_CHECKSUM_LEN)},
+                {TAG_KEY_LONG_NAME, StringTk::generateRandomS3TagValue(TAG_KEY_LONG_NAME,
+                                                                       TAG_VALUE_LONG_LEN,
+                                                                       TAG_CHECKSUM_LEN)}
+        };
+
+        for (const auto& pair : keyValuePairs)
+        {
+            tag.WithKey(pair.first).SetValue(pair.second);
+            tagging.AddTagSet(tag);
+        }
+
+        request.SetBucket(bucketName);
+        request.SetTagging(tagging);
+
+        S3::PutBucketTaggingOutcome taggingOutcome = s3Client->PutBucketTagging(request);
+        if (!taggingOutcome.IsSuccess())
+        {
+            const auto &s3Error = taggingOutcome.GetError();
+
+            throw WorkerException(std::string("Bucket tagging failed. ") +
+                                  "Endpoint: " + s3EndpointStr + "; " +
+                                  "Bucket: " + bucketName + "; " +
+                                  "Exception: " + s3Error.GetExceptionName() + "; " +
+                                  "Message: " + s3Error.GetMessage());
+        }
+
+        if (doS3TagInlineVerify)
```

Comment:
this would probably be the place to add and immediately afterwards remove the "ghost tag", if that's still planned